### PR TITLE
Only perform update/delete when a primary key is present

### LINKED ISSUE #9: Getting exception "Expected 1 row affected, actual 4." when deleting RecordSet with multiple Records
I'm not sure if this belongs in the ORM project, the Mapper project, or this one, so I'm filing it in the project where the exception was thrown.

I've got a simple table setup, no relationships. Running a select query to find all records with a given user ID and email address, then deleting them all. Code looks like this:

```php
        $recordSet = $this->atlas
            ->select(TableName::class)
            ->where('user_id =', $userId)
            ->andWhere('email_address =', $emailAddress)
            ->fetchRecordSet();

        $recordSet->setDelete();

        $atlas->persistRecordSet($recordSet);
```

When I run it, it deletes all the matching records as expected, but it throws an exception coming from https://github.com/atlasphp/Atlas.Table/blob/faeec8db3e2dacb432144563cd405b17600f4e6e/src/Table.php#L252

that reads "Expected 1 row affected, actual 4." in an example that has 4 matching rows.

Not sure what's going on here, but it looks like `$pdoStatement` is grabbing the count for all affected rows and for some reason the caller expects there to only be 1 affected row.

### PR DESCRIPTION
**Pull request by pmjones**
Fixes #9

### CODE REVIEW

File: src/Exception.php [hunk 1]
```diff
@@ -33,36 +33,48 @@ public static function invalidType(string $expect, $actual) : Exception
         return new Exception("Expected type $expect; got $actual instead.");
     }
 
-    public static function unexpectedRowCountAffected($count)
+    public static function unexpectedRowCountAffected($count) : Exception
```
Comment (froschdesign) [inline src/Exception.php]: The class is not final but the method declaration is changed. Are you planning a new major release or what does your backwards compatibility promise say in this case?

File: src/Table.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ public function __construct(
         $this->queryFactory = $queryFactory;
         $this->tableEvents = $tableEvents;
         $this->rowClass = substr(static::CLASS, 0, -5) . 'Row';
-        if (count($this::PRIMARY_KEY) == 1) {
+        if (count($this::PRIMARY_KEY) < 2) {
             $this->primaryKey = new PrimarySimple($this::PRIMARY_KEY[0]);
```
Comment (kevinsmith) [inline src/Table.php]: When I don't have a primary key, I get an "Undefined offset: 0" here.

Comment (froschdesign) [reply src/Table.php]: The problem here: missing test cases for these changes.

**Comment by kevinsmith** [comment]
This takes care of it! Got an exception when there wasn't a primary key, and the operation was successful when a PK was present.

**Comment by pmjones** [comment]
@kevinsmith Sweet. I'll try to wrap up a new release today.

**Pull request closed without merging**